test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -132,66 +133,130 @@ var _ = Describe("Mixedcpus", Ordered, func() {
-				withRuntime(components.GetComponentName(profile.Name, components.ComponentNamePrefix)))
-			Expect(err).ToNot(HaveOccurred())
-			cmd := printMixedCPUsEnvCmd()
-			output, err := pods.ExecCommandOnPod(testclient.K8sClient, p, "", cmd)
-			Expect(err).ToNot(HaveOccurred(), "failed to execute command on pod; cmd=%q pod=%q", cmd, client.ObjectKeyFromObject(p).String())
-			isolatedAndShared := strings.Split(string(output), "\r\n")
-			isolated, err := cpuset.Parse(isolatedAndShared[0])
-			Expect(err).ToNot(HaveOccurred())
-			Expect(isolated.IsEmpty()).ToNot(BeTrue())
-			shared, err := cpuset.Parse(isolatedAndShared[1])
-			ppShared, _ := cpuset.Parse(string(*profile.Spec.CPU.Shared))
-			Expect(err).ToNot(HaveOccurred())
-			Expect(shared.Equals(ppShared)).To(BeTrue(), "OPENSHIFT_SHARED_CPUS value not equal to what configure in the performance profile."+
-				"OPENSHIFT_SHARED_CPUS=%s spec.cpu.shared=%s", shared.String(), ppShared.String())
+
+		When("Modifying the shared CPUs in the performance profile", func() {
+			BeforeEach(func() {
+
+				isolated := mustParse(string(*profile.Spec.CPU.Isolated))
+				shared := mustParse(string(*profile.Spec.CPU.Shared))
+				// select one arbitrary
+				isolatedCore := mustParse(strconv.Itoa(isolated.List()[0]))
+				sharedCore := mustParse(strconv.Itoa(shared.List()[0]))
+				// swap one from the isolated with one from the shared
+				//  to create new shared cpuset
+				newIsolated := isolated.Difference(*isolatedCore).Union(*sharedCore)
+				newShared := shared.Difference(*sharedCore).Union(*isolatedCore)
+				profile.Spec.CPU.Isolated = cpuSetToPerformanceCPUSet(&newIsolated)
+				profile.Spec.CPU.Shared = cpuSetToPerformanceCPUSet(&newShared)
+				By("applying new performanceProfile")
+				testprofiles.UpdateWithRetry(profile)
+				mcp, err := mcps.GetByProfile(profile)
+				Expect(err).ToNot(HaveOccurred())
+				By("waiting for mcp to catch up")
+				mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+				mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+				Expect(testclient.Client.Get(ctx, client.ObjectKeyFromObject(profile), profile))
+				testlog.Infof("new isolated CPU set=%q\nnew shared CPU set=%q", newIsolated.String(), newShared.String())
+				// we do not bother to revert the profile at the end of the test, since its irrelevant which of the cpus are shared
+			})
+
+			It("should contains the updated values under the container", func() {
+				rl := &corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("1"),
+					corev1.ResourceMemory: resource.MustParse("100Mi"),
+					sharedCpusResource:    resource.MustParse("1"),
+				}
+				p, err := createPod(ctx, testclient.Client, testutils.NamespaceTesting,
+					withRequests(rl),
+					withLimits(rl),
+					withRuntime(components.GetComponentName(profile.Name, components.ComponentNamePrefix)))
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
this is a good check but it is similar to the feature first configuration and verification. I think what even adds more value in the update test is to verify that a controller-related pod (e.g. deployment pod) gets updated with the new shared cpuset, given that an orphan pod will not survive node reboot.

Reply:
What about the following flow:
1. create deployment with pod asks for shared cpus
2. check shared cpus reflected under the pod
3. change the shared cpus under the performance profile
4. check that the pod from step 1 reflect the new shared cpus

Is this type of test work for you?